components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java

```diff
@@ -1020,4 +1022,70 @@ private String extractPropertyName(String pathProp) {
+        if (StringUtils.isBlank(userStoreName)) {
+            throw handleException(Response.Status.BAD_REQUEST, UserStoreConstants.ErrorMessage.
+                    ERROR_CODE_INVALID_INPUT);
+        }
+        try {
+            UserStoreConfigService userStoreConfigService = UserStoreConfigServiceHolder.getInstance()
+                    .getUserStoreConfigService();
+            classNames = userStoreConfigService.getAvailableUserStoreClasses();
+            if (CollectionUtils.isEmpty(classNames) ||
+                    !classNames.contains(userStoreName)) {
+                throw handleException(Response.Status.NOT_FOUND, UserStoreConstants.ErrorMessage.
+                        ERROR_CODE_NOT_FOUND);
+            }
+        } catch (IdentityUserStoreMgtException e) {
+            UserStoreConstants.ErrorMessage errorEnum =
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_ERROR_RETRIEVING_USER_STORE;
+            throw handleIdentityUserStoreMgtException(e, errorEnum);
+        }
+        UserStoreAttributeMapping userStoreAttributeMapping = new UserStoreAttributeMapping();
+        List<UserStoreAttributeDO> attributeMappings = getAttributeMappings(userStoreName);
+        userStoreAttributeMapping = userStoreAttributeMapping
+                .attributeMapping(attributeMappings)
+                .typeId(typeId)
+                .typeName(userStoreName);
+        try {
+            boolean isLocal = UserStoreManagerRegistry.isLocalUserStore(userStoreName);
+            userStoreAttributeMapping.setIsLocal(isLocal);
+        } catch (UserStoreException e) {
+            LOG.error(String.format("Userstore type is not found for %s", userStoreName), e);
+        }
+        return userStoreAttributeMapping;
+    }
+
+    /**
+     * Get user store attribute mappings for a given user store typeId.
+     *
+     * @param userStoreName String user store name (base64 decoded user store id).
+     * @return List of user store attribute mappings for the given typeId.
+     */
+    private List<UserStoreAttributeDO> getAttributeMappings(String userStoreName) {
+
+        UserStoreConfigService userStoreConfigService = UserStoreConfigServiceHolder.getInstance().
+                getUserStoreConfigService();
+        try {
+            UserStoreAttributeMappings userStoreAttributeMappings = userStoreConfigService.
+                    getUserStoreAttributeMappings();
+            Map<String, UserStoreAttributeDO> mapping = userStoreAttributeMappings
+                    .getUserStoreAttributeMappings(userStoreName);
+            return new ArrayList<>(mapping.values());
+        } catch (IdentityUserStoreMgtException e) {
+            LOG.error("Error occurred while retrieving user store attribute metadata", e);
+            throw handleException(Response.Status.INTERNAL_SERVER_ERROR, UserStoreConstants.ErrorMessage.
+                    ERROR_CODE_ERROR_RETRIEVING_USER_STORE_ATTRIBUTE_METADATA);
```

Comment:
As a practice, we do not log and throw the exception because it will print the stack trace twice. We do not log unless it is the last place in the chain. So here I guess we do not need the log. WDYT?

Reply:
We use the `handleException()` function to throw APIErrors.  `handleException()` function does not accept `Exception e` as a parameter. Hence I think we should log the exception here. There will be no duplicate stack traces.